File: Arcade/Intro/Java/ChessKnight.java
final class ChessKnight {

	boolean isSafe(int x, int y) {
	    return x >= 0 && x < 8 && y >= 0 && y < 8;
	}

	int chessKnight(String cell) {
	    int x = cell.charAt(0) - 'a',
	        y = new Integer(cell.charAt(1)+"") - 1,
	        out = 0;
	    
	    if(isSafe(x + 2, y + 1)) out++;
	    if(isSafe(x + 2, y - 1)) out++;
	    if(isSafe(x - 2, y + 1)) out++;
	    if(isSafe(x - 2, y - 1)) out++;
	    if(isSafe(x + 1, y + 2)) out++;
	    if(isSafe(x - 1, y + 2)) out++;
	    if(isSafe(x + 1, y - 2)) out++;
	    if(isSafe(x - 1, y - 2)) out++;
	    
	    return out;
	}
}

File: Arcade/Intro/Ruby/chessKnight.rb
def isSafe(x, y)
    return (x >= 0 and x < 8 and y >= 0 and y < 8);
end

def chessKnight(cell)
    x, y, out = cell[0].ord - 97, cell[1].to_i - 1, 0
    
    out += 1 if isSafe(x + 2, y + 1);
    out += 1 if isSafe(x + 2, y - 1);
    out += 1 if isSafe(x - 2, y + 1);
    out += 1 if isSafe(x - 2, y - 1);
    out += 1 if isSafe(x + 1, y + 2);
    out += 1 if isSafe(x - 1, y + 2);
    out += 1 if isSafe(x + 1, y - 2);
    out += 1 if isSafe(x - 1, y - 2);
    
    return out;
end